### DISCUSSION TITLE
Should I put x-data in div?

### DISCUSSION THREAD

**Discussion by bubnenkoff** [alpinejs/alpine#2492] [1. Help] [answered]
I am looking an examples. 100% of them are showing of usage `x-data` inside `div`. But do I really need it? For example this code:
```
        <div class="tabs right-align">
            <a>Tab 1</a>
            <a>Tab 2</a>
            <a>Tab 3</a>
             <a x-data="{ active: false }">
                    <button @click="active = !active">
                        <span>Login</span>
                            <div x-show="active" class="dropdown">
                                <a>
                                <label>Complementary text</label>
                                </a>
                            </div>                         
                    </button>
                </a>
        </div>
``` 
I placed `x-data` in `a` tag. All works fine.

But I also could do:
```
<button x-data="{ active: false }" @click="active = !active">
```
Is it's correct to place data and action on same level?

**Comment by danddanddand** [comment]
It's personal preference. If you think about browser API everything is just inheriting Element (https://developer.mozilla.org/en-US/docs/Web/API/Element) as a base class and Alpine isn't opinionated about it. 

It's whatever makes sense to you and your use case.

**Comment by felixfrey** [comment, marked as answer]
> Is it's correct to place data and action on same level?

Yes you can do this, its correct!
